src/plugins/intel_cpu/tests/unit/shape_inference_test/custom_shape_infer/matmul.cpp

```diff
@@ -0,0 +1,149 @@
+            std::transform(a_shape.cbegin(),
+                           a_shape.cend() - 2,
+                           b_shape.cbegin(),
+                           std::back_inserter(exp_shape),
+                           [](const StaticDimension& a, const StaticDimension& b) {
+                               return std::max(a.get_length(), b.get_length());
+                           });
+            exp_shape.push_back(*std::next(a_shape.rbegin()));
+            exp_shape.push_back(b_shape.back());
+        } else if (a_shape.size() == 1 && b_shape.size() > 1) {
+            exp_shape = b_shape;
+            exp_shape.erase(std::prev(exp_shape.end(), 2));
+        } else if (b_shape.size() == 1 && a_shape.size() > 1) {
+            exp_shape = a_shape;
+            exp_shape.erase(std::prev(exp_shape.end()));
+        }
+    }
+
+    static StaticShape make_transpose_input(const StaticShape& in) {
+        StaticShape out(in);
+        if (out.size() > 1) {
+            std::iter_swap(out.rbegin(), std::next(out.rbegin()));
+        }
+        return out;
+    }
+
+    StaticShape a_shape, b_shape, exp_shape;
+    unit_test::ShapeVector output_shapes;
+};
+
+TEST_P(CPUMatMulTest, no_input_transpose) {
+    const auto matmul = make_matmul(a_shape.size(), b_shape.size(), false, false);
+
+    std::vector<StaticShape> static_input_shapes = {a_shape, b_shape};
+
+    GTEST_SKIP() << "Skipping test, please check CVS-108946";
+    // TODO  108946,below test case can't pass
+    unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
+}
+
+TEST_P(CPUMatMulTest, transpose_input_a) {
+    const auto matmul = make_matmul(a_shape.size(), b_shape.size(), true, false);
+
+    const auto a_transpose = make_transpose_input(a_shape);
+    std::vector<StaticShape> static_input_shapes = {a_transpose, b_shape};
+
+    GTEST_SKIP() << "Skipping test, please check CVS-108946";
+    // TODO 108946,below test case can't pass
+    unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
+}
+
+TEST_P(CPUMatMulTest, transpose_input_b) {
+    const auto matmul = make_matmul(a_shape.size(), b_shape.size(), false, true);
+
+    const auto b_transpose = make_transpose_input(b_shape);
+    std::vector<StaticShape> static_input_shapes = {a_shape, b_transpose};
+
+    GTEST_SKIP() << "Skipping test, please check CVS-108946";
+    // TODO 108946,below test case can't pass
+    unit_test::cpu_test_shape_infer(matmul.get(), static_input_shapes, output_shapes);
+}
+
+TEST_P(CPUMatMulTest, transpose_inputs_a_b) {
+    const auto matmul = make_matmul(a_shape.size(), b_shape.size(), true, true);
+
+    const auto a_transpose = make_transpose_input(a_shape);
+    const auto b_transpose = make_transpose_input(b_shape);
+
+    std::vector<StaticShape> static_input_shapes = {a_transpose, b_transpose};
+
+    GTEST_SKIP() << "Skipping test, please check CVS-108946";
+    // TODO 108946,below test case can't pass
```

Comment:
May be it does make sense to put the skip at the beginnig? Here and in other test definitions?

Reply:
fixed